tests/drivers/spi/spi_loopback/boards/mimxrt1064_evk.overlay

```diff
@@ -5,14 +5,86 @@
  */
 
 &lpspi1 {
-	slow@0 {
+	lpspi_slow: slow@0 {
 		compatible = "test-spi-loopback-slow";
 		reg = <0>;
 		spi-max-frequency = <500000>;
 	};
-	fast@0 {
+	lpspi_fast: fast@0 {
 		compatible = "test-spi-loopback-fast";
 		reg = <0>;
 		spi-max-frequency = <16000000>;
 	};
 };
+
+&pinctrl {
+	pinmux_flexio3spi0: pinmux_flexio3spi0 {
+		group0 {
+			pinmux =
+				<&iomuxc_gpio_ad_b0_03_gpio1_io03>, /* cs */
+				<&iomuxc_gpio_ad_b1_10_flexio3_flexio10>, /* sck */
+				<&iomuxc_gpio_ad_b1_01_flexio3_flexio01>, /* sdo */
+				<&iomuxc_gpio_ad_b1_04_flexio3_flexio04>; /* sdi */
+			drive-strength = "r0-6";
+			slew-rate = "slow";
+			nxp,speed = "150-mhz";
+		};
+	};
+	pinmux_flexio3spi1: pinmux_flexio3spi1 {
+		group0 {
+			pinmux =
+				<&iomuxc_gpio_ad_b0_02_gpio1_io02>, /* cs */
+				<&iomuxc_gpio_ad_b1_11_flexio3_flexio11>, /* sck */
+				<&iomuxc_gpio_ad_b1_00_flexio3_flexio00>, /* sdo */
+				<&iomuxc_gpio_ad_b1_05_flexio3_flexio05>; /* sdi */
+			drive-strength = "r0-6";
+			slew-rate = "slow";
+			nxp,speed = "150-mhz";
+		};
+	};
+};
+
+&flexio3 {
+	status = "okay";
+	flexio3_spi0: flexio3_spi0 {
+		compatible = "nxp,flexio-spi";
+		status = "okay";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		cs-gpios = <&gpio1 3 GPIO_ACTIVE_LOW>;
+		sdo-pin = <1>;
+		sdi-pin = <4>;
+		sck-pin = <10>;
+		pinctrl-0 = <&pinmux_flexio3spi0>;
+		pinctrl-names = "default";
+		flexio_spi_slow: slow@0 {
+			status = "disabled";
+			compatible = "test-spi-loopback-slow";
+			reg = <0>;
+			spi-max-frequency = <500000>;
+		};
+	};
+	flexio3_spi1: flexio3_spi1 {
+		compatible = "nxp,flexio-spi";
+		status = "okay";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		cs-gpios = <&gpio1 2 GPIO_ACTIVE_LOW>;
+		sdo-pin = <0>;
+		sdi-pin = <5>;
+		sck-pin = <11>;
+		pinctrl-0 = <&pinmux_flexio3spi1>;
+		pinctrl-names = "default";
+		flexio_spi_fast: fast@0 {
+			status = "disabled";
+			compatible = "test-spi-loopback-fast";
+			reg = <0>;
+			spi-max-frequency = <4000000>;
+		};
+	};
+};
```

Comment:
Why are the above lines added to this overlay file. Should they be in the `flexio-overlay` file

Reply:
@mmahadevan108, I made it so that the test could be run on the board. If a successful build is enough for testing, then in the `flexio-overlay` file it possible configure only `flexio1` without pinctrl.

Reply:
@MikhailSiomin, can we move the flexio changes out of`boards/mimxrt1064_evk.overlay` file and move them to `overlay-mcux-flexio-spi.overlay`. We can use the build command:` west build -p -b mimxrt1064_evk zephyr/tests/drivers/spi/spi_loopback -T drivers.spi.mcux_flexio_spi.loopback` to pull the right overlay file when testing for FlexIO.
With the current changes, FlexIO gets enabled when building for the default spi_loopback test which is incorrect.